src/V4/PeanutV4.3.sol

```diff
@@ -592,24 +797,29 @@ contract PeanutV4 is IERC721Receiver, IERC1155Receiver, ReentrancyGuard {
         require(_index < deposits.length, "DEPOSIT INDEX DOES NOT EXIST");
         Deposit memory _deposit = deposits[_index];
         require(_deposit.claimed == false, "DEPOSIT ALREADY WITHDRAWN");
-
-        // Compute the hash of the withdrawal message
-        bytes32 _recipientAddressHash = ECDSA.toEthSignedMessageHash(
-            keccak256(
-                abi.encodePacked(
-                    PEANUT_SALT,
-                    block.chainid,
-                    address(this),
-                    _index,
-                    _recipientAddress,
-                    _extraData
+        
+        // check that the signer is the same as the one stored in the deposit.
+        // Signature may be empty for address-bound deposits.
+        address depositSigner;
+        if (_signature.length > 0) {
+            // Compute the hash of the withdrawal message
+            bytes32 _recipientAddressHash = ECDSA.toEthSignedMessageHash(
+                keccak256(
+                    abi.encodePacked(
+                        PEANUT_SALT,
+                        block.chainid,
+                        address(this),
+                        _index,
+                        _recipientAddress,
+                        _extraData
+                    )
                 )
-            )
-        );
-        // check that the signer is the same as the one stored in the deposit
-        address depositSigner = getSigner(_recipientAddressHash, _signature);
+            );
+            depositSigner = getSigner(_recipientAddressHash, _signature);
+        }
         require(!_deposit.requiresMFA || _authorized, "REQUIRES AUTHORIZATION");
-        require(depositSigner == _deposit.pubKey20, "WRONG SIGNATURE");
+        require(_deposit.pubKey20 == address(0) || depositSigner == _deposit.pubKey20, "WRONG SIGNATURE");
```

Comment:
nit: this could be done more efficiently by moving it to line 803 and having an if else there
`require(_deposit.pubKey20 == address(0) || depositSigner == _deposit.pubKey20, "WRONG SIGNATURE");`

Reply:
I disagree. I think it's better when security checks are one-lined and happen all together rather than in different places in different `if` conditions, so since it's a "nit", won't implement.